### DISCUSSION TITLE
middleware/mysql/faq/faq-others

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#5241] [Announcements]
# middleware/mysql/faq/faq-others

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/middleware/mysql/faq/faq-others.html

**Comment by san3Xian** [comment]
由于某些原因 dce5 mysql主从需要铲掉重建，但是删除mysql pvc删除mysql pod后，mysql pod一直在不断崩溃重启

mysql主从需要铲掉重建，但是删除mysql pvc删除mysql pod后，mysql pod一直在崩溃重启,是因为 rpl_semi_sync_master_enabled 和 rpl_semi_sync_slave_enabled 和 rpl_semi_sync_master_wait_for_slave_count 参数不能在 mysql 初始化的时候使用 (这个报错只会在mysql容器第一次启动的时候打出来，后面的崩溃就会提示类似mysql表不存在的报错)

可以按照以下方法处理(configmap 的名字自行根据需要修改的 mysql 实例进行替换)

kubectl scale --replicas 0 sts -n mcamel-system mysql-operator #将 mysql 脱离operator管控防止 mysql cm被改回来
kubectl edit cm -n mcamel-system mcamel-common-kpanda-mysql-cluster-mysql # 把 rpl_semi_sync_master_enabled 和 rpl_semi_sync_slave_enabled 和 rpl_semi_sync_master_wait_for_slave_count 之类的配置都删掉

如果嫌弃kubectl edit cm排版乱的话，可以参考下述操作：
configmap中的内容

换行前不要有空白字符
```
:%s/\\s\\n/\\n/g
```
没有tab键(\t)
```
:%s/\\t/    /g
```
文尾没有换行符和空白字符(整个data object的最后一个字符不能是\n)
即可解决kubectl edit / get cm 排版混乱问题



第三步:

删除 mysql pvc 删除 mysql pod，这时候mysql pod0会回来，可能整个pod计数重启1-2次，但是不会一直崩溃，然后处于3/4

然后就把operator scale回来，接着应该就好了

kubectl scale --replicas 3 sts -n mcamel-system mysql-operator
